File: Level2/줄 서는 방법.py
from itertools import permutations
import math


def solution(n, k):
    lis = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20]
    target = lis[:n]
    answer = []

    for i in range(n):
        gap = math.factorial(n - (i + 1))
        answer.append(target[(k - 1) // gap])
        del target[(k - 1) // gap]
        print(answer)

    return answer
print(solution(3, 5))

File: Level2/줄 서는 방법.java
import java.util.*;
class Solution {
    public int[] solution(int n, long k) {
        int count = 0;
        int [] answer = new int[n];
        ArrayList<Integer> arr = new ArrayList<>();
        long factorial = 1L;
        for(int i=1; i<=n; i++) {
            factorial *= i;
            arr.add(i);
        }
        k--;
        while(n > 0) {
            factorial /= n;
            int value = (int) (k / factorial);
            answer[count++] = arr.get(value);
            arr.remove(value);
            k %= factorial;
            n--;
        }
        
        return answer;
    }
}
